homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,486 @@
+"""Switches for AVM Fritz!Box functions."""
+from __future__ import annotations
+
+import logging
+
+from fritzconnection.core.exceptions import FritzConnectionException, FritzSecurityError
+import xmltodict
+
+from homeassistant.components.network.util import async_get_source_ip
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .common import FritzBoxBaseSwitch, FritzBoxTools, SwitchInfo
+from .const import (
+    DOMAIN,
+    SWITCH_TYPE_DEFLECTION,
+    SWITCH_TYPE_DEVICEPROFILE,
+    SWITCH_TYPE_PORTFORWARD,
+    SWITCH_TYPE_WIFINETWORK,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def service_call_action(
+    fritzbox_tools: FritzBoxTools,
+    service_name: str,
+    service_suffix: bool,
+    action_name: str,
+    **kwargs,
+) -> None | dict:
+    """Return service details."""
+
+    service_append = "1" if service_suffix else ""
```

Comment:
If you already know "1" needs to be appended, call the function with f"{service_name}1" instead